internal/scheduler/metrics/metrics.go

```diff
@@ -315,7 +316,11 @@ func (m *Metrics) UpdateLeased(job *jobdb.Job, jctx *schedulercontext.JobSchedul
 		job = jctx.Job.(*jobdb.Job)
 	}
 	latestRun := job.LatestRun()
-	priorState, priorStateTime := getPriorState(job, latestRun, latestRun.LeaseTime())
+	leaseTime := time.Now()
```

Comment:
Can we instead set the lease time on the job from within the scheduler when creating the lease?